pkg/run/install/install_flux.go

```diff
@@ -3,53 +3,109 @@ package install
 import (
 	"context"
 	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"strings"
 
 	coretypes "github.com/weaveworks/weave-gitops/core/server/types"
 	"github.com/weaveworks/weave-gitops/pkg/flux"
 	"github.com/weaveworks/weave-gitops/pkg/logger"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetFluxVersion(ctx context.Context, log logger.Logger, kubeClient client.Client) (string, error) {
-	log.Actionf("Getting Flux version ...")
-
-	listResult := unstructured.UnstructuredList{}
+// Heuristic mapping from the Source controller version to the Flux version.
+// We support back to v0.32 only and the guess will be only the main version number.
+// For example, Flux v0.38.x versions contain the Source controller v0.33.0,
+// but we will report the Flux version as v0.38.0 only.
+//
+// Source: https://github.com/fluxcd/flux2/releases
+//
+// How to update this map:
+// 1. Find the Flux version you want to support in the URL above.
+// 2. Find the Source controller version that is used in that Flux version
+// 3. Add the new mapping to the map below
+// 4. Don't forget to *remove the oldest mapping* from the map below
+var sourceVerToFluxVer = map[string]string{
```

Comment:
⚠️ We'd have to bump this map every time a new Flux version is released which is error-prone. Did you check whether the OLM-deployed source-controller Deployment has the `app.kubernetes.io/version` label? This should directly point to the Flux version.

Reply:
Unfortunately, it doesn't have any version label there.
Also we don't have that kind of label neither on Azure.

Let's discuss what would be the alternative approaches.

Reply:
Aw man, that's too bad. Ok, thanks for looking into it.